src/client/shader.cpp

```diff
@@ -469,7 +469,7 @@ u32 ShaderSource::getShaderIdDirect(const std::string &name,
 	infostream<<"getShaderIdDirect(): "
 			<<"Returning id="<<id<<" for name \""<<name<<"\""<<std::endl;
 
-	return id;
+	return info.shader_failed ? 0 : id;
```

Comment:
Suggestion: Add the error message here. `if (info.shader_failed) client->setFatalError("Failed to compile " + name + "! Check the log for details.");`

Reply:
I would have to pass the client pointer to ShaderSource, which seems like it would be a mess :/